src/main/java/io/jenkins/docker/pipeline/DockerAgent.java

```diff
@@ -0,0 +1,32 @@
+package io.jenkins.docker.pipeline;
+
+import hudson.Extension;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.pipeline.modeldefinition.agent.DeclarativeAgent;
+import org.jenkinsci.plugins.pipeline.modeldefinition.agent.DeclarativeAgentDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+@SuppressWarnings("unchecked") // TODO DeclarativeAgent.getDescriptor problem
+public class DockerAgent extends DeclarativeAgent<DockerAgent> {
+
+    private static final long serialVersionUID = 1;
+
+    public final String image;
+
+    @DataBoundConstructor
+    public DockerAgent(String image) {
+        this.image = image;
+    }
+
+    // TODO other properties accepted by DockerNodeStep
```

Comment:
This feels pretty incomplete to me without this `TODO` being implemented. Not sure I would be comfortable supporting this when it is so incomplete.